crates/utils/src/rate_limit/mod.rs

```diff
@@ -38,14 +35,14 @@ pub struct BucketConfig {
 #[derive(Clone)]
 pub struct RateLimit {
   configs: Arc<RwLock<EnumMap<ActionType, BucketConfig>>>,
-  backends: EnumMap<ActionType, InMemoryBackend<RateLimitIpAddr>>,
+  backends: EnumMap<ActionType, LemmyBackend>,
```

Comment:
have you considered instead of creating many different backends for each action, you could use a single backend, and include the action inside the rate limit key?
This could potentially save a little bit of memory, and you would only need 1 GC task.
I guess it is a tradeoff between an extra byte in the keys vs overhead of multiple maps...